### PR TITLE
`checked_shr` implementation and coverage

### LINKED ISSUE #18: [Feature]: `checked_shr` function
Implement a function that perform a right bitwise shift with overflow checking.

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #18 

Move Coverage: 96.19 

## Summary by CodeRabbit

* **New Features**
  * Introduced checked right shift operations across all integer types (u8, u16, u32, u64, u128, u256). These operations return `None` if any non-zero bits would be discarded during the shift, ensuring no silent precision loss occurs.

* **Tests**
  * Added comprehensive test coverage validating the new checked right shift behavior, including edge cases and overflow scenarios.

### CODE REVIEW

**Comment by bidzyyys** [comment]
Blocked by https://github.com/OpenZeppelin/contracts-sui/pull/32

File: math/core/sources/u128.move [hunk 1]
```diff
@@ -16,3 +16,12 @@ public fun mul_div(a: u128, b: u128, denominator: u128, rounding_mode: RoundingM
         (false, result as u128)
     }
 }
+
+/// Shift the value right by the given number of bits.
+/// Returns `None` if the shift consumes a non-zero bit.
```
Comment (immrsd) [inline math/core/sources/u128.move]: Should we also mention that `None` will be returned as well for the case when `shift` value is >= than the number of bits in a type?

**Review by immrsd — APPROVED** [review]
LGTM
